File: DS/linked list/arrToLinkedList.js
// node of a linked list
class Node{
    constructor(info){
        this.link = null;
        this.info = info;
    }
    getInfo(){
        return this.info;
    }
    getLink(){
        return this.link;
    }
    setLink(link){
        this.link = link;
    }
}
// given root/head element, it inserts new node at the end of a linked list
const insert = function(root, item){
    const temp = new Node(item);
    let current;

    if(root==null)
        root=temp;
    else{
        current = root;
        while(current.getLink()!=null) {
            current = current.link;
        }
        current.setLink(temp);
    }    
    return root;
}

// prints linked list start from root
const display = function(root){
    let current = root;
    while(current!=null){
        console.log(current.getInfo());
        current = current.link;
    }
}

// arr -> Linked List
const arrToList = function(arr){
    let root = null;
    for(let i=0 ; i<arr.length ; i++)
        root = insert(root, arr[i]);   
    return root;     
}

const arr = [1,2,3,4,5,6,7,8,9];
const root = arrToList(arr);

display(root);

File: DS/linked list/arrToLinkedList.cpp
// O(n*n

#include <iostream>
using namespace std;

class Node{
    public:
        int data;
        Node *link;
    Node(int val){
        data = val;
        link = NULL;
    }    
};

Node *insert(Node *root, int item){
    Node *newNode = new Node(item);

    if (root==NULL)
        root = newNode;
    else{
        Node *current = root;
        while(current->link!=NULL){
            current = current->link;
        }
        current->link = newNode;
    }    
    return root;
}

void display(Node* root) 
{ 
    while (root != NULL) { 
        cout << root->data << " "; 
        root = root->link; 
    } 
} 

Node *arrayToList(int arr[], int n) 
{ 
    Node *root = NULL; 
    for (int i = 0; i < n; i++) 
        root = insert(root, arr[i]); 
   return root; 
} 

int main(){
    int arr[] = { 1, 2, 6, 4, 5 }; 
    int n = sizeof(arr) / sizeof(arr[0]); 
    Node* root = arrayToList(arr, n); 
    display(root); 
    return 0;
}
